packages/core/demo/demo-data/bar.ts

```diff
@@ -69,6 +69,23 @@ export const groupedBarOptions = {
 	}
 };
 
+// Horizontal Grouped
+export const groupedHorizontalBarData = groupedBarData;
+
+export const groupedHorizontalBarOptions = {
+	title: "Grouped horizontal bar (discrete)",
+	axes: {
+		left: {
+			scaleType: "labels",
+			primary: true,
+		},
+		bottom: {
+			secondary: true
+		}
+	},
+	orientation: "horizontal"
```

Comment:
What happens if I flip the scales but don't add the `orientation` property?

Maybe this is covered elsewhere, but what's the function of the `primary`/`secondary` properties?

Reply:
The `primary` & `secondary` fields are a bit confusing right now...

We basically need 1 of each, and the flipping of the 2 would do nothing...

I WOULD like to rethink that portion

Reply:
Okay ... I won't block the PR on that, but maybe something to make an issue for?

Reply:
So currently, you'd have to set `orientation` to `horizontal` then make bottom scale linear and left axis `labels` based.

Shouldn't that scale change be detected to switch the bars around?

Reply:
That's what I would expect yeah ... Internally we can track it with an `orientation` enum, but externally it probably makes more sense if it just does the "right thing"

Reply:
i wanted the chart to work horizontally regardless of how the user maps the axes. 
for example this **makes sense** :

```

export const simpleHorizontalBarOptions = {
	title: "Simple horizontal bar (discrete)",
	axes: {
		left: {
			primary: true,
			scaleType: "labels"
		},
		bottom: {
			secondary: true,
		}
	},
	orientation: "horizontal"
};
```
 but this (below) does not really make sense, but can provide a chart nonetheless. 

```
export const simpleHorizontalBarOptions = {
	title: "Simple horizontal bar (discrete)",
	axes: {
		left: {
			primary: true,
		},
		bottom: {
			secondary: true,
			scaleType: "labels"
		}
	},
	orientation: "horizontal"
};
```

you would get this chart
![image](https://user-images.githubusercontent.com/14351335/70647926-019bc500-1c18-11ea-9c81-a6df37ee6a04.png)


Although it seems smart to let carbon-charts "decide" whether to go horizontal/vertical it seems to me that it would negatively affect some of the customizability that we introduced in the refactor.
By boasting that our refactor makes custom charts easy to create (even if they are not to spec and nor reccommended) I think we shouldn't decide for the users (or we provide a configuration to force a desired orientation).  I guess im proposing that if we want charts to "decide" that we keep a configuration for orientation which can be provided to force an orientation.

Reply:
I don't think we would ruin the customization by simplifying _how_ we declare the chart orientation ... In your example the chart isn't what I would expect as a user, given that I could very easily miss the `orientaton` property - if we let the data drive the layout of the chart (in this case, the axes config data) we can actually be _more_ customizable, and clearer in how the customization works